From 54f5bf1670d4ab5c1c27d9b3cc6d61fa316f5b4f Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 1 Sep 2021 14:05:42 +0200
Subject: [PATCH 132/841] feat(stm32mp2): update boot API file

Align with BootROM DV18 boot_api.h file.
The file has not changed between DV18 and DV23.
Put boot_action field usage under MP13/MP15 flags as it is no more used
on MP25.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 4ea83fb7d9f69b5af03877fb231141dfb193c2b7 feat(stm32mp2): update boot API file]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: I48f749b13e01c265315225398220e9a4db0c9ece
---
 plat/st/common/include/stm32mp_common.h |  2 ++
 plat/st/common/stm32mp_common.c         |  6 ++++
 plat/st/stm32mp2/include/boot_api.h     | 47 -------------------------
 3 files changed, 8 insertions(+), 47 deletions(-)

diff --git a/plat/st/common/include/stm32mp_common.h b/plat/st/common/include/stm32mp_common.h
index f2d296795..a90a48fea 100644
--- a/plat/st/common/include/stm32mp_common.h
+++ b/plat/st/common/include/stm32mp_common.h
@@ -21,7 +21,9 @@
 void stm32mp_save_boot_ctx_address(uintptr_t address);
 uintptr_t stm32mp_get_boot_ctx_address(void);
 uint16_t stm32mp_get_boot_itf_selected(void);
+#if STM32MP13 || STM32MP15
 uint32_t stm32mp_get_boot_action(void);
+#endif
 
 bool stm32mp_is_single_core(void);
 bool stm32mp_is_closed_device(void);
diff --git a/plat/st/common/stm32mp_common.c b/plat/st/common/stm32mp_common.c
index 72c394de2..3c3813661 100644
--- a/plat/st/common/stm32mp_common.c
+++ b/plat/st/common/stm32mp_common.c
@@ -71,7 +71,9 @@ unsigned int plat_get_syscnt_freq2(void)
 
 static uintptr_t boot_ctx_address;
 static uint16_t boot_itf_selected;
+#if STM32MP13 || STM32MP15
 static uint32_t boot_action_saved;
+#endif
 
 void stm32mp_save_boot_ctx_address(uintptr_t address)
 {
@@ -79,7 +81,9 @@ void stm32mp_save_boot_ctx_address(uintptr_t address)
 
 	boot_ctx_address = address;
 	boot_itf_selected = boot_context->boot_interface_selected;
+#if STM32MP13 || STM32MP15
 	boot_action_saved = boot_context->boot_action;
+#endif
 }
 
 uintptr_t stm32mp_get_boot_ctx_address(void)
@@ -92,10 +96,12 @@ uint16_t stm32mp_get_boot_itf_selected(void)
 	return boot_itf_selected;
 }
 
+#if STM32MP13 || STM32MP15
 uint32_t stm32mp_get_boot_action(void)
 {
 	return boot_action_saved;
 }
+#endif
 
 uintptr_t stm32mp_ddrctrl_base(void)
 {
diff --git a/plat/st/stm32mp2/include/boot_api.h b/plat/st/stm32mp2/include/boot_api.h
index d08194d05..d3bed7631 100644
--- a/plat/st/stm32mp2/include/boot_api.h
+++ b/plat/st/stm32mp2/include/boot_api.h
@@ -18,53 +18,6 @@
  * Boot Context related definitions
  */
 
-/*
- * Possible value of boot context field 'boot_action'
- */
-/* Boot action is Process Secure Boot */
-#define BOOT_API_CTX_BOOT_ACTION_SECURE_BOOT_PROCESS		0x04U
-/* Boot action is Process Wakeup from D1STANDBY or STANDBY */
-#define BOOT_API_CTX_BOOT_ACTION_WAKEUP_D1STANDBY_OR_STANDBY	0x05U
-/* Boot action is Process DEV_BOOT */
-#define BOOT_API_CTX_BOOT_ACTION_DEV_BOOT			0x06U
-
-#define BOOT_API_CTX_BOOT_ACTION_LOCAL_C1_RESET_PROCESS		0x0AU
-
-/*
- * Possible value of boot context field 'wakeup_status'
- */
-
-/* The boot reason is not a D1Standby Exit reason */
-#define BOOT_API_CTX_NO_D1STANDBY_NO_STANDBY_EXIT		0x00
-
-/*
- * The boot reason is a D1Standby without previous Standby
- * (VDDCORE was kept and SYSRAM content preserved)
- * and the CPU1 is TDCID
- */
-#define BOOT_API_CTX_D1STANDBY_EXIT_NO_STANDBY_CPU1_TDCID	0x01
-
-/*
- * The boot reason is a D1Standby without previous Standby
- * (VDDCORE was kept and SYSRAM content preserved)
- * and the CPU1 is Not TDCID (ie CPU2 is TDCID)
- */
-#define BOOT_API_CTX_D1STANDBY_EXIT_NO_STANDBY_CPU1_NOT_TDCID	0x02
-
-/*
- * The boot reason is a D1Standby with previous Standby
- * (VDDCORE was Off and SYSRAM content lost)
- * and the CPU1 is TDCID
- */
-#define BOOT_API_CTX_D1STANDBY_EXIT_STANDBY_CPU1_TDCID		0x03
-
-/*
- * The boot reason is a D1Standby with previous Standby
- * (VDDCORE was Off and SYSRAM content was lost)
- * and the CPU1 is Not TDCID (ie CPU2 is TDCID)
- */
-#define BOOT_API_CTX_D1STANDBY_EXIT_STANDBY_CPU1_NOT_TDCID	0x04
-
 /*
  * Possible value of boot context field 'auth_status'
  */
-- 
2.47.2

